docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,112 @@
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When c8y-log-plugin is started, it will publish health status as `up` on to `tedge/health/c8y-log-plugin` topic
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management) to the c8y cloud.
+
+Pros: 
+-   This will remove the file system dependency when the daemons want to create the supported operation files on the mapper file system.
+    For example, tedge-agent wants to create the supported operations files in /etc/tedge/operations/c8y directory.
+
+Cons: 
+-   This leads to hard dependency on the list of daemons in the mapper. For example, the tedge-mapper-c8y has to subscribe to tedge/health/tedge-agent
+topic to receive the health status from the tedge-agent. Based on this the mapper has to create a supported operation file in /etc/tedge/operations/c8y.
+-   This will be a hard-coded dependency of the name of the plugin and can’t be updated on the fly if someone wants to change the name of the daemon
+    or wants to add a new daemon.
+
+
+    ## Proposal 2:  Introduce an `init message` mechanism
+
+    ```admonish note
+    Pre-condition for this proposal is that the `plugins` must be `cloud-agnostic` and the mappers will be `cloud-specific`.
+    The plugins will be grouped based on the roles they perform. For example, software-management, device-restart, config-management, log-management, firmware-management,.etc.
+    ```
+
+    The tedge plugins must publish a `retained init` message describing their role (software-management, config management, etc)
+     and other details specific to that role.
+
+    The proposed topic will be `tedge/<device-id>/init/<cloud>`.
+    Here `<cloud>` component is `optional` and could be c8y, azure, aws, etc
+
+    The message template is {Operation_name: <content>, Operation_name:<content>, Configs: type1..typen}.
+    Here `Operation_name` is the name of the operation to be published to the cloud and 
+    the `content` is the content of the operation file, if any.
+    Here `Configs` is the other configuration information that is to be passed to the cloud.
+    For Example, in case of `log-plugin` the `Configs` is log types that are supporeted.
+
+    Example 1:
+    - When the tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
```

Comment:
```suggestion
    - When the tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init` and `tedge/<device-id>/init/c8y` topics
```
As I don't see why the tedge mapper would be interested in a `#` subscription that includes init messages meant for other clouds as well.